### PR TITLE
add recipe for czmq

### PR DESCRIPTION
**Pull request by vchuravy**

### CODE REVIEW

File: C/czmq/build_tarballs.jl [hunk 1]
```diff
@@ -0,0 +1,48 @@
...
+    --enable-drafts \
+    --disable-static \
+    CXXFLAGS="-O2 -fms-extensions"
+make -j${nproc}
+make install
+"""
+
+# These are the platforms we will build for by default, unless further
+# platforms are passed in on the command line
+platforms = expand_cxxstring_abis(supported_platforms())
```
Comment (giordano) [inline C/czmq/build_tarballs.jl:36]: Is it necessary to expand the C++ string ABI?

File: C/czmq/build_tarballs.jl [hunk 2]
```diff
@@ -0,0 +1,48 @@
...
+
+# The products that we will ensure are always built
+products = [
+    LibraryProduct("libczmq", :libczmq),
+]
+
+# Dependencies that must be installed before this package can be built
+dependencies = Dependency[
+    Dependency("ZeroMQ_jll", v"4.3")
```
Comment (giordano) [inline C/czmq/build_tarballs.jl:45]: You need to be more specific, we have 4.3.1 and 4.3.2

Comment (vchuravy) [reply C/czmq/build_tarballs.jl:45]: Hm I want to say any "4.3" series.

File: C/czmq/build_tarballs.jl (hunk 1 above)
Comment (vchuravy) [inline C/czmq/build_tarballs.jl:36]:
```suggestion
platforms = supported_platforms()
```

File: C/czmq/build_tarballs.jl (hunk 2 above)
Comment (vchuravy) [inline C/czmq/build_tarballs.jl:45]:
```suggestion
    Dependency("ZeroMQ_jll", v"4.3.2")
```

**Pull request merged**